plugins/csp-web-api/src/Plugin.cpp

```diff
@@ -466,18 +468,27 @@ void Plugin::update() {
         }
 
       } else {
-        // Capturing color images is pretty straight-forward.
-        std::vector<std::byte> capture(mCaptureWidth * mCaptureHeight * 3);
-        glReadPixels(0, 0, mCaptureWidth, mCaptureHeight, GL_RGB, GL_UNSIGNED_BYTE, &capture[0]);
-
-        if (mCaptureFormat == "tiff") {
-          tiffWriteToVector(mCapture, capture, mCaptureWidth, mCaptureHeight, 3, 8);
-        } else if (mCaptureFormat == "png") {
-          stbi_write_png_to_func(&stbWriteToVector, &mCapture, mCaptureWidth, mCaptureHeight, 3,
-              capture.data(), mCaptureWidth * 3);
+        if (mCaptureFormat == "raw") {
+          // larger output size, but performance is comparable or better because there is no image encoding
+          mCapture.resize(mCaptureWidth * mCaptureHeight * 3 * 4);
+          std::shared_ptr<cs::graphics::HDRBuffer> hdrBuffer = mGraphicsEngine->getHDRBuffer();
+          VistaTexture* luminance_buffer = hdrBuffer->getCurrentWriteAttachment();
+          luminance_buffer->Bind(GL_TEXTURE_2D);
+          glGetTexImage(GL_TEXTURE_2D, 0, GL_RGB, GL_FLOAT, (void*)mCapture.data());
```

Comment:
This produces `GL_INVALID_ENUM error generated. <texture> exceeds the maximum supported number of combined texture image units.` since it tries to bind the texture to the texture unit at 33984 :smile: 

```suggestion
          luminance_buffer->Bind();
          glGetTexImage(GL_TEXTURE_2D, 0, GL_RGB, GL_FLOAT, (void*)mCapture.data());
          luminance_buffer->Unbind();
```